WordPress/Classes/ViewRelated/Reader/Subscriptions/ReaderSubscriptionsView.swift

```diff
@@ -128,9 +129,57 @@ struct ReaderSubscriptionsView: View {
         ReaderSubscriptionHelper().unfollow(site)
     }
 
-    private func reloadSearchResults(searchText: String) {
-        let ranking = StringRankedSearch(searchTerm: searchText)
-        searchResults = ranking.search(in: subscriptions) { "\($0.title) \($0.siteURL)" }
+    private func performBackgroundSearch(searchText: String) {
+        guard !searchText.isEmpty else {
+            searchResults = nil
+            pendingSearchText = nil
+            searchTask?.cancel()
+            return
+        }
+
+        guard searchTask == nil else {
+            pendingSearchText = searchText
+            return
+        }
+
+        searchTask = Task { [searchText] in
+            await performSearch(for: searchText)
+        }
+    }
+
+    private func performSearch(for searchText: String) async {
+        let searchableData = subscriptions.map(SearchableSubscription.init)
+
+        let resultObjectIDs = await StringRankedSearch(searchTerm: searchText)
+            .parallelSearch(in: searchableData) { $0.searchableText }
+            .map(\.objectID)
+
+        searchResults = subscriptions.filter { resultObjectIDs.contains($0.objectID) }
```

Comment:
Convert `resultObjectIDs` to a `Set` before `filter`?

Reply:
Oh right forgot to do it – fixed.